include/deal.II/matrix_free/cuda_matrix_free.templates.h

```diff
@@ -504,6 +504,29 @@ namespace CUDAWrappers
       if (cell < gpu_data.n_cells)
         func(cell, &gpu_data, &shared_data, src, dst);
     }
+
+
+
+    template <int dim, typename Number, typename functor>
+    __global__ void
+    evaluate_coeff(functor                                      func,
```

Comment:
Why is a copy preferable here?

Reply:
Same reason as above. Basically if the function is `__global__` it means that the function is "called" from the cpu but runs on the gpu and you can't pass by reference. So if the function is `__global__`, you need to pass by copy. If the function is `__device__` you can do what you want because you stay on the gpu.